roles/callhome/precheck/tasks/check.yml

```diff
@@ -0,0 +1,69 @@
+---
+- block:
+  - name: callhome_precheck | Check if callhome is enabled.
+    assert:
+     that:
+     - callhome_params.is_enabled is defined
+     - callhome_params.is_enabled | bool
+     fail_msg: "Callhome is disabled"
+     success_msg: "Callhome is enabled. Proceeding to check if customer information has been supplied."
+    when:  ansible_hostname  == callhome_params.callhome_server
+
+  - name: callhome_precheck |  Check if customer information has been supplied.
+    assert:
+     that:
+     - callhome_params.customer_name is defined
+     - callhome_params.customer_name| length > 0
+     - callhome_params.customer_email is defined
+     - callhome_params.customer_email| length > 0
+     - callhome_params.customer_id is defined
+     - callhome_params.customer_id| int > 0
+     - callhome_params.customer_country is defined
+     - callhome_params.customer_country| length > 0
+     fail_msg: "Callhome customer information is not defined."
+     success_msg: "Customer information is defined. Proceeding to check if proxy is enabled"
+    when: ansible_hostname  ==  callhome_params.callhome_server
+
+  - name: callhome_precheck | Check if proxy is enabled
+    assert:
+     that:
+     - callhome_params.proxy_ip is defined
+     - callhome_params.proxy_ip|bool
+     - callhome_params.proxy_port is defined
+     - callhome_params.proxy_port|bool
+     fail_msg: "Proxy is not enabled. Proceeding to perform network check"
+     success_msg: "Proxy is enabled. Proceeding to check if proxy user and password is defined"
+    ignore_errors: yes
+    when: ansible_hostname  ==  callhome_params.callhome_server
+    register: scale_callhome_proxy_status
+    failed_when: false
+
+  - name: callhome_precheck | Check if proxy user and passsword is defined
+    assert:
+     that:
+     - callhome_params.proxy_user is defined
+     - callhome_params.proxy_user|bool
+     - callhome_params.proxy_password is defined
+     - callhome_params.proxy_password|bool
+     fail_msg: "Proxy user and password not defined. It is mandatory to define proxy user and password if proxy ip and port is defined"
+     success_msg: "Proxy information is defined.Proceeding to perform network check"
+    when: ansible_hostname  ==  callhome_params.callhome_server and
+          callhome_params.proxy_ip is defined and
+          callhome_params.proxy_port is defined and
+          callhome_params.proxy_ip|bool and
+          callhome_params.proxy_port|bool
+    register: scale_callhome_proxy_status
+
+  - name: callhome_precheck | Checking network connectivity for callhome configuration.
+    uri:
+     url: https://esupport.ibm.com
+     timeout: 10
+     return_content: yes
+    when: ansible_hostname  ==  callhome_params.callhome_server
+
+  - debug:
+     msg: "Callhome precheck OK"
+    when: ansible_hostname  == callhome_params.callhome_server
+    #- debug:
+    # msg: "{{ inventory_hostname }}"
+     #run_once: true

```

Comment:
Should delete the code that is not being used.  If this is something that we need to help with debug, we should probably add it into one of the name fields ... 

```
- name: "doing something for: {{ inventory_hostname }}" 
```

And then would not require a recreate if we hit issues in this code.